frontend/public/components/secrets/create-secret/utils.ts

```diff
@@ -94,3 +97,38 @@ export const useSecretDescription = (typeAbstraction: SecretTypeAbstraction): st
       return null;
   }
 };
+
+export const opaqueEntriesToObject = (opaqueEntriesArray: OpaqueDataEntry[]): SecretStringData => {
+  return _.reduce(
+    opaqueEntriesArray,
+    (acc, entry) =>
+      _.assign(acc, {
+        [entry?.key]: entry?.isBase64 || entry?.isBinary ? entry.value : Base64.encode(entry.value),
+      }),
+    {},
+  );
+};
+
+export const newOpaqueSecretEntry = (): OpaqueDataEntry => {
+  return {
+    key: '',
+    value: '',
+    uid: _.uniqueId(),
+  };
+};
+
+export const opaqueSecretObjectToArray = (stringData: SecretStringData): OpaqueDataEntry[] => {
+  if (_.isEmpty(stringData)) {
+    return [newOpaqueSecretEntry()];
+  }
+  return _.map(stringData, (value, key) => {
+    const isValueBinary = isBinary(null, Buffer.from(value, 'base64'));
+    return {
+      key,
+      value: isValueBinary ? Base64.encode(value) : value,
+      isBase64: isValueBinary,
+      isBinary: isValueBinary,
```

Comment:
I think we only need one of these flags:
```suggestion
      isBinary: isValueBinary,
```